include/seastar/core/shared_mutex.hh

```diff
@@ -234,6 +237,46 @@ with_lock(shared_mutex& sm, Func&& func) noexcept {
     }
 }
 
+/// \brief Construct a RAII-based shared lock corresponding to a given Mutex object.
+///
+/// Since constructors cannot be exectued asynchronously, use this function
+/// to shared lock a Mutex and construct an std::shared_lock object
+/// that will unlock the mutex when it's being destroyed.
+///
+/// The caller is responsible for keeping the corresponding Mutex object
+/// alive at least until the returned lock is destroyed.
+template <typename Mutex>
```

Comment:
nitpick: I'm not sure the name "Mutex" really represents the type of object this is. It needs to have exactly the right methods lock_shared(), unlock_shared(). Would have been nice to create a concept like https://en.cppreference.com/w/cpp/named_req/SharedLockable (just with a different return type for lock_shared()) and use that concept.
But it's a nitpick, not critical.

Reply:
That's a good idea. I'll add the concepts